src/main/java/org/dcsa/tnt/service/impl/TNTEventSubscriptionTOServiceImpl.java

```diff
@@ -0,0 +1,357 @@
+    } else {
+      eventTypes = eventSubscriptionTO.getEventType();
+    }
+
+    return Flux.fromIterable(eventTypes)
+        .concatMap(
+            eventType ->
+                eventSubscriptionRepository.insertEventTypeForSubscription(
+                    eventSubscriptionTO.getSubscriptionID(), eventType))
+        .then(Mono.just(eventSubscriptionTO));
+  }
+
+  private Mono<Void> createTransportDocumentEventTypes(TNTEventSubscriptionTO eventSubscriptionTO) {
+
+    if (null == eventSubscriptionTO.getTransportDocumentTypeCode()) {
+      eventSubscriptionTO.setTransportDocumentTypeCode(ALL_TRANSPORT_DOCUMENT_TYPES);
+    }
+
+    List<TransportDocumentTypeCode> transportDocumentTypeCodes =
+        eventSubscriptionTO.getTransportDocumentTypeCode();
+    return Flux.fromIterable(transportDocumentTypeCodes)
+        .flatMap(
+            td ->
+                eventSubscriptionRepository.insertTransportDocumentEventTypeForSubscription(
+                    eventSubscriptionTO.getSubscriptionID(), td))
+        .then();
+  }
+
+  private Mono<Void> createShipmentEventType(TNTEventSubscriptionTO eventSubscriptionTO) {
+
+    if (null == eventSubscriptionTO.getShipmentEventTypeCode()) {
+      eventSubscriptionTO.setShipmentEventTypeCode(ALL_SHIPMENT_EVENT_TYPES);
+    }
+
+    List<ShipmentEventTypeCode> shipmentEventTypeCode =
+        eventSubscriptionTO.getShipmentEventTypeCode();
+
+    return Flux.fromIterable(shipmentEventTypeCode)
+        .flatMap(
+            s ->
+                eventSubscriptionRepository.insertShipmentEventTypeForSubscription(
+                    eventSubscriptionTO.getSubscriptionID(), s))
+        .then();
+  }
+
+  private Mono<Void> createTransportEventType(TNTEventSubscriptionTO eventSubscriptionTO) {
+
+    if (null == eventSubscriptionTO.getTransportEventTypeCode()) {
+      eventSubscriptionTO.setTransportEventTypeCode(ALL_TRANSPORT_EVENT_TYPES);
+    }
+
+    List<TransportEventTypeCode> transportEventTypeCode =
+        eventSubscriptionTO.getTransportEventTypeCode();
+    return Flux.fromIterable(transportEventTypeCode)
+        .flatMap(
+            t ->
+                eventSubscriptionRepository.insertTransportEventTypeForSubscription(
+                    eventSubscriptionTO.getSubscriptionID(), t))
+        .then();
+  }
+
+  private Mono<Void> createEquipmentEventType(TNTEventSubscriptionTO eventSubscriptionTO) {
+
+    if (null == eventSubscriptionTO.getEquipmentEventTypeCode()) {
+      eventSubscriptionTO.setEquipmentEventTypeCode(ALL_EQUIPMENT_EVENT_TYPES);
+    }
+
+    List<EquipmentEventTypeCode> equipmentEventTypeCode =
+        eventSubscriptionTO.getEquipmentEventTypeCode();
+
+    return Flux.fromIterable(equipmentEventTypeCode)
+        .flatMap(
+            e ->
+                eventSubscriptionRepository.insertEquipmentEventTypeForSubscription(
+                    eventSubscriptionTO.getSubscriptionID(), e))
+        .then();
+  }
```

Comment:
Put in EventCore - same argument as above - we might need it in the future (DCSA_EVENT_HUB)